### PR TITLE
Prettify printing of tests

### PR DESCRIPTION
**Pull request by albinahlback**
Examples:
```
gr_mat_window_init_clear                          0.10   (PASS)
gr_mat_solve_triu
    947 success,     6 domain,    47 unable       0.16   (PASS)
gr_poly_hgcd                                      0.63   (PASS)
perm_parity                                       0.04   (PASS)
flint_ctz                                         0.01   (PASS)
flint_fprintf                                     0.00   (PASS)
flint_printf                                          (SKIPPED)
d_is_nan                                          0.00   (PASS)
fmpz_poly_hensel_lift_without_only_inverse        1.06   (PASS)
fmpz_mod_poly_compose_mod_brent_kung_precomp_preinv_threaded
                                                  0.50   (PASS)
fmpq_dedekind_sum                                 1.92   (PASS)
```
Here `PASS` is bold green and `SKIPPED` is bold yellow. It would also be nice to standardize how failures would be treated, but then one would have to go through the whole codebase.

One can disable colored tests by pushing `--disable-pretty-tests` to configure.

The number before `PASS` reflects the time it took to run the test. I haven't specified the unit as it is just using `CLOCKS_PER_SEC` to determine the "time" it took. It is more of a guideline of how long the test took.

Basically, the format I was looking for was `FUNCTION  TIME  RESULT`. Ideally, one would like to first print the function it tests (for which one `printf` call is made), then perform tests, and finally print time and status (another `printf` call). However, when running multiple processes at a time, this is problematic as function and time-result typically appear on different lines. Hence, for now I just chose to print when result is obtained. Don't know how it works for multiple processes, but I suppose one could use something like `ncurses`, but that would perhaps be a little overkill.

Thinking of it, during failures it will not print what test failed, unless specified in error message. But this would perhaps only motivate us to unify fails in tests? ;-)

### CODE REVIEW

**Comment by fredrik-johansson** [comment]
> Ideally, one would like to first print the function it tests (for which one printf call is made), then perform tests, and finally print time and status (another printf call). However, when running multiple processes at a time, this is problematic as function and time-result typically appear on different lines. Hence, for now I just chose to print when result is obtained.

The problem with this is when a test hangs, and you don't see which one it is.

**Comment by albinahlback** [comment]
> > Ideally, one would like to first print the function it tests (for which one printf call is made), then perform tests, and finally print time and status (another printf call). However, when running multiple processes at a time, this is problematic as function and time-result typically appear on different lines. Hence, for now I just chose to print when result is obtained.
> 
> The problem with this is when a test hangs, and you don't see which one it is.

Yeah. Let me think of a solution...

**Comment by albinahlback** [comment]
Okay, I think the only viable option is to print something like
```
FUNCTION...
FUNCTION    TIME  RESULT
```
This will take up double the number of lines when doing tests, but I think it is worth it.

Of course, if we knew that `make check` was called single processed, it would be trivial.

**Comment by fredrik-johansson** [comment]
> Of course, if we knew that make check was called single processed, it would be trivial.

I thought you already used some special magic to handle ``make -j``, meaning that you already check for this?

**Comment by albinahlback** [comment]
> > Of course, if we knew that make check was called single processed, it would be trivial.
> 
> I thought you already used some special magic to handle `make -j`, meaning that you already check for this?

I meant that oneliners are trivial (at least on Linux) for singleprocessed testing as we can delete a line by pushing `\r`. But for multiprocessed testing we cannot do that anymore. And I would prefer if the output is the same for both.

**Comment by albinahlback** [comment]
Will merge when tests pass.

**Comment by albinahlback** [comment]
Btw, I increased the range of the randomization of some tests in `ulong_extras`, as well as making them à la Fredrik (like you did with aliasing for the `fmpz` tests etc.).

**Comment by fredrik-johansson** [comment]
Nice!
